topic/timeseries/exploratory_data_analysis.ipynb

```diff
@@ -0,0 +1,935 @@
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>timestamp</th>\n",
+       "      <th>location</th>\n",
+       "      <th>temperature</th>\n",
+       "      <th>humidity</th>\n",
+       "      <th>wind_speed</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>1672532724000</td>\n",
+       "      <td>Vienna</td>\n",
+       "      <td>20.329405</td>\n",
+       "      <td>90.035633</td>\n",
+       "      <td>7.045718</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1</th>\n",
+       "      <td>1672533002000</td>\n",
+       "      <td>Dornbirn</td>\n",
+       "      <td>20.485222</td>\n",
+       "      <td>90.233420</td>\n",
+       "      <td>4.534103</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>1672535400000</td>\n",
+       "      <td>Vienna</td>\n",
+       "      <td>21.331066</td>\n",
+       "      <td>74.014164</td>\n",
+       "      <td>4.347653</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>3</th>\n",
+       "      <td>1672536651000</td>\n",
+       "      <td>Berlin</td>\n",
+       "      <td>21.034967</td>\n",
+       "      <td>35.431482</td>\n",
+       "      <td>13.160233</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>1672537835000</td>\n",
+       "      <td>Dornbirn</td>\n",
+       "      <td>20.801286</td>\n",
+       "      <td>34.073989</td>\n",
+       "      <td>5.314130</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "       timestamp  location  temperature   humidity  wind_speed\n",
+       "0  1672532724000    Vienna    20.329405  90.035633    7.045718\n",
+       "1  1672533002000  Dornbirn    20.485222  90.233420    4.534103\n",
+       "2  1672535400000    Vienna    21.331066  74.014164    4.347653\n",
+       "3  1672536651000    Berlin    21.034967  35.431482   13.160233\n",
+       "4  1672537835000  Dornbirn    20.801286  34.073989    5.314130"
+      ]
+     },
+     "execution_count": 12,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "query = \"SELECT * FROM weather_data\"\n",
+    "with engine.connect() as conn:\n",
+    "    result = conn.execute(sa.text(query))\n",
+    "    columns = result.keys() # Extract column names\n",
+    "    df = pd.DataFrame(result.fetchall(), columns=columns)\n",
+    "df.head(5)"
+   ]
```

Comment:
This fails to execute, because SQLAlchemy 2.0.0 creates incompatibility with Pandas read_sql (I ended up with issue explained in https://github.com/pandas-dev/pandas/issues/51015)